### PR TITLE
diff: exit with correct status

### PR DESCRIPTION
**Pull request by andrewliebenow**

### CODE REVIEW

File: text/diff_util/constants.rs [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #![allow(dead_code)]
 
-pub const EXIT_STATUS_NO_DIFFERENCE: i32 = 0;
-pub const EXIT_STATUS_DIFFERENCE: i32 = 1;
-pub const EXIT_STATUS_TROUBLE: i32 = 2;
+pub const EXIT_STATUS_NO_DIFFERENCE: u8 = 0_u8;
+pub const EXIT_STATUS_DIFFERENCE: u8 = 1_u8;
+pub const EXIT_STATUS_TROUBLE: u8 = 2_u8;
```
Comment (jgarzik) [inline text/diff_util/constants.rs]: Don't add type tags where the type is specified in the same line.   Just creates visual redundancies, and the redundancies do not improve code generation.

**Review by jgarzik — CHANGES_REQUESTED** [review]
Mostly looks OK.  Minor nit commented.

File: text/diff_util/constants.rs (hunk 1 above)
Comment (andrewliebenow) [reply text/diff_util/constants.rs]: While I was at it, I switched the tests to use these constants instead of the enum, since that seemed a bit circular.

**Review by jgarzik — APPROVED** [review]

**Comment by jgarzik** [comment]
Unfortunately, post-merge, a megaton of new build warnings appeared on macos (and probably linux also), so had to revert.

**Pull request merged**